schema/nlog.schema.json

```diff
@@ -0,0 +1,559 @@
+            }
+          ],
+          "additionalProperties": true
+        },
+        "rules": {
+          "oneOf": [
+            {
+              "$id": "#/NLog/properties/rules",
+              "type": "array",
+              "title": "rules",
+              "description": "",
+              "default": [],
+              "examples": [
+                [
+                  {
+                    "logger": "*",
+                    "minLevel": "Trace",
+                    "writeTo": "all-file, database"
+                  },
+                  {
+                    "logger": "Microsoft.*",
+                    "maxLevel": "Info",
+                    "final": true
+                  }
+                ]
+              ],
+              "additionalItems": true,
+              "items": {
+                "$id": "#/NLog/properties/rules/item",
+                "type": "object",
+                "title": "logger rule",
+                "description": "Send events to targets",
+                "default": {},
+                "examples": [
+                  {
+                    "logger": "*",
+                    "minLevel": "Error",
+                    "writeTo": "database"
+                  },
+                  {
+                    "logger": "*",
+                    "minLevel": "Trace",
+                    "writeTo": "all-file, database",
+                    "enabled": "false"
+                  },
+                  {
+                    "logger": "MyNamespace.MyClass",
+                    "minLevel": "Trace",
+                    "writeTo": "MyClass-file"
+                  }
+                ],
+                "required": [
+                  "logger"
+                ],
+                "additionalProperties": true,
+                "properties": {
+                  "logger": {
+                    "$id": "#/NLog/properties/rules/item/properties/logger",
+                    "type": "string",
+                    "title": "logger",
+                    "description": "Filter on the name of the logger. May include wildcard characters ('*' or '?').",
+                    "examples": [
+                      "*",
+                      "Microsoft.*",
+                      "*.Something.*"
+                    ]
+                  },
+                  "ruleName": {
+                    "$id": "#/NLog/properties/rules/item/properties/rulename",
+                    "type": "string",
+                    "title": "rulename",
+                    "description": "Rule identifier to allow rule lookup with Configuration.FindRuleByName and Configuration.RemoveRuleByName.",
+                    "examples": [
+                      "*"
+                    ]
+                  },
+                  "level": {
+                    "$id": "#/NLog/properties/rules/item/properties/level",
+                    "type": "string",
+                    "title": "level",
+                    "description": "",
+                    "enum": [
```

Comment:
Will `enum` not restrict the available options? Since you can use layout-renderers now.